python/ray/util/spark/databricks_hook.py

```diff
@@ -63,12 +73,69 @@ def on_ray_dashboard_created(self, port):
             get_spark_session().sparkContext, port, "Ray Cluster Dashboard"
         )
 
-    def on_spark_background_job_created(self, job_group_id):
+    def on_cluster_created(self, ray_cluster_handler):
+        db_api_entry = get_dbutils().entry_point
         try:
-            get_dbutils().entry_point.registerBackgroundSparkJobGroup(job_group_id)
+            db_api_entry.registerBackgroundSparkJobGroup(
+                ray_cluster_handler.spark_job_group_id
+            )
         except Exception:
             _logger.warning(
                 "Register ray cluster spark job as background job failed. You need to "
                 "manually call `ray_cluster_on_spark.shutdown()` before detaching "
                 "your databricks python REPL."
             )
+
+        auto_shutdown_timeout_minutes = float(
+            os.environ.get(DATABRICKS_RAY_ON_SPARK_AUTOSHUTDOWN_TIMEOUT_MINUTES, "30")
+        )
```

Comment:
What happens if a user sets `DATABRICKS_RAY_ON_SPARK_AUTOSHUTDOWN_TIMEOUT_MINUTES` to a negative value (e.g. -1) by mistake?